app/models/concerns/user_subscription_sourceable.rb

```diff
@@ -4,7 +4,7 @@ module UserSubscriptionSourceable
   # This all assumes there's an association with User under the column user_id.
 
   included do
-    has_many :user_subscriptions, as: :user_subscription_sourceable
+    has_many :user_subscriptions, as: :user_subscription_sourceable, dependent: :destroy
```

Comment:
I added this in this PR as it's the only model using the `user_subscriptions`. Those should go if the article goes away

Reply:
Should we have a warning about this? I feel like if I collected a bunch of emails from an article and then decided to remove that article I might be a little caught off guard when my list of emails also suddenly disappears. @atsmith813 any thoughts?

Reply:
I don't think we want to destroy subscriptions. While we don't currently support displaying subscriptions for a deleted article in a view for an author, we still would want to retain the data. The subscriptions could still have value to the author, even on a deleted Article. In the future, we can show the subscriptions for _Deleted Article_ when we introduce more author analytics.

Here are some more thoughts on this: https://github.com/forem/forem/pull/9405/files/b01d5fec24bd3b07c48f23b4b2e5d3891896219c#diff-41409b89978eea2120acf9882a9b2ccf

Reply:
Ah ah I totally forgot that conversation and apparently never replied to your last question.

We currently have a "null: false" on the column, so there's no possibility for subscriptions to sit in the DB with `user_subscription_sourceable_id` as null.

I think we should switch to your proposal here:

>  I wonder if there's a quick win to validate the presence of user_subscription_sourceable only on create 🤔 . A subscription shouldn't be created without it, but it can be updated to be removed.

What do you both think?

Reply:
I'm down with that approach 💯 .

Reply:
Ok, I'm going to set this back to draft and send a PR to change that

Reply:
https://github.com/forem/forem/pull/10033